flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -447,6 +453,165 @@ public void postStop() throws Exception {
 		return CompletableFuture.completedFuture(Acknowledge.get());
 	}
 
+	@Override
+	public CompletableFuture<Acknowledge> rescaleJob(
+			int newParallelism,
+			RescalingBehaviour rescalingBehaviour,
+			Time timeout) {
+		final ArrayList<JobVertexID> allOperators = new ArrayList<>(jobGraph.getNumberOfVertices());
+
+		for (JobVertex jobVertex : jobGraph.getVertices()) {
+			allOperators.add(jobVertex.getID());
+		}
+
+		return rescaleOperators(allOperators, newParallelism, rescalingBehaviour, timeout);
+	}
+
+	@Override
+	public CompletableFuture<Acknowledge> rescaleOperators(
```

Comment:
Does it have to be part of the `JobMasterGateway` interface? It's only used from `rescaleJob`.

Reply:
Yes it will be necessary, once we do rescaling for individual operators. This will be the case with the rescaling policies.

Reply:
ok